engine/crates/integration-tests/src/federation/request/stream.rs

```diff
@@ -0,0 +1,91 @@
+use std::future::IntoFuture;
+
+use bytes::Bytes;
+use engine::BatchRequest;
+use futures::{future::BoxFuture, stream::BoxStream, StreamExt, TryStreamExt};
+use gateway_core::StreamingFormat;
+use headers::HeaderMapExt;
+
+pub struct MultipartStreamRequest(pub(super) super::ExecutionRequest);
+
+impl MultipartStreamRequest {
+    pub async fn collect<B>(self) -> B
+    where
+        B: Default + Extend<serde_json::Value>,
+    {
+        self.await.stream.collect().await
+    }
+}
+
+impl IntoFuture for MultipartStreamRequest {
+    type Output = GraphqlStreamingResponse;
+
+    type IntoFuture = BoxFuture<'static, Self::Output>;
+
+    fn into_future(self) -> Self::IntoFuture {
+        let mut headers = self.0.http_headers();
+        headers.typed_insert(StreamingFormat::IncrementalDelivery);
+        let request = BatchRequest::Single(self.0.request.into_engine_request());
+        Box::pin(async move {
+            let response = self.0.engine.execute(headers, request).await;
+            let stream = multipart_stream::parse(response.body.into_stream().map_ok(Into::into), "-")
+                .map(|result| serde_json::from_slice(&result.unwrap().body).unwrap());
```

Comment:
:zap: **Suggestion** 
The `unwrap()` calls in the `MultipartStreamRequest` and `SseStreamRequest` implementations can lead to panics if the `Result` is an `Err`. Consider handling the error more gracefully, perhaps by using `?` to propagate the error or by providing a default value.